src/transformers/pipelines/text_classification.py

```diff
@@ -156,7 +156,7 @@ def __call__(self, *args, **kwargs):
         result = super().__call__(*args, **kwargs)
         # TODO try and retrieve it in a nicer way from _sanitize_parameters.
         _legacy = "top_k" not in kwargs
-        if isinstance(args[0], str) and _legacy:
+        if ("inputs" in kwargs or isinstance(args[0], str)) and _legacy:
```

Comment:
```suggestion
        if (len(args) > 1 and isinstance(args[0], str)) or isinstance(kwargs.get("inputs", None), str) and _legacy:
```

And we do need a small test proving the new call actually works.

Reply:
Unfortunately, I don't know exactly why _legacy checks if ``top_k`` is in kwargs since I'm a relatively new user of this library so I can't say if we could do this check in other way.

With respect to the  ``(len(args) > 1 and isinstance(args[0], str))`` check,

Finally, with respect to the two checks of "Inputs" and the check of legacy ``_legacy``, should we use parentheses around the ``or``? ``True or True and False = True`` and ``(True or True) and False = False`` although I guess the first two bools can't be true simultaneously because the ``TypeError: Pipeline.__call__() got multiple values for argument 'inputs'`` error.

Reply:
Also, I tested 3 different scenarios but I'm not sure if the last one returns what we would expect:
```python
# Case 1
print(classifier("Hello world args"))
print(classifier(["Hello world args"]))

print(80 * '=')

# Case 2
print(classifier("Hello world args"  , top_k=2))
print(classifier(["Hello world args"], top_k=2))

print(80 * '=')

# Case 3
print(classifier("Hello world args", 2))
print(classifier(["Hello world args"], 2))
```
1. A dict containing the label with higher score and a list with one dict containing the label with higher score: OK
```
 {'label': 'POSITIVE', 'score': 0.9990410208702087}
[{'label': 'POSITIVE', 'score': 0.9990410208702087}]
```
2. OK. Same case than the previous one but with two dicts containing the two higher scores.
```
 [{'label': 'POSITIVE', 'score': 0.9990410208702087}, {'label': 'NEGATIVE', 'score': 0.0009589608525857329}]
[[{'label': 'POSITIVE', 'score': 0.9990410208702087}, {'label': 'NEGATIVE', 'score': 0.0009589608525857329}]]
```
3. OK? I think that this case should return the same that the case 1 since it's ignoring ``args[1]=2``. However, in this case, it returns a list with one dict containing the label with higher score:
```
Ignoring args : (2,)
Ignoring args : (2,)
[{'label': 'POSITIVE', 'score': 0.9990410208702087}]
Ignoring args : (2,)
Ignoring args : (2,)
[{'label': 'POSITIVE', 'score': 0.9990410208702087}]
```

Reply:
@Narsil could you confirm if the case 3 is OK? If it's ok, I would just add some tests as you pointed.

Reply:
Making sure everything is OK is pretty time consuming and the reason why I try not to handle this sugar when it's not absolutely required.

I'll defer to @ArthurZucker on this one.

Reply:
Ok, I understand. Thanks for your time!